Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -12,12 +12,18 @@ public class IonObjectSerializer : IonSerializer<object>
         private readonly IonSerializer ionSerializer;
         private readonly IonSerializationOptions options;
         private readonly Type targetType;
+        private readonly IEnumerable<PropertyInfo> readOnlyProperties;
 
         public IonObjectSerializer(IonSerializer ionSerializer, IonSerializationOptions options, Type targetType)
         {
             this.ionSerializer = ionSerializer;
             this.options = options;
             this.targetType = targetType;
+
+            if (!this.options.IgnoreReadOnlyProperties && !this.options.IncludeFields)
```

Comment:
Shouldn't this read the properties only if you are including fields?

Reply:
So if we are serializing/deserializing read-only properties then we need to serialize/deserialize the property's backing field (and that is what `this.readOnlyProperties` is used for).
However the `IncludeFields` option results in all fields (including backing fields) being included which eliminates any need for special logic to detect those backing fields.